netlogo-gui/src/main/agent/AgentSet.scala

```diff
@@ -34,6 +34,10 @@ abstract class AgentSet(
   def randomSubsetGeneral(resultSize: Int, precomputedCount: Int, rng: api.MersenneTwisterFast): Array[Agent]
   def toLogoList: LogoList
 
+  // this is an upper bound on the size of the agentset. It's used in heuristics, so it should
+  // be free or cheap to compute.
+  private[agent] def sizeBound: Int
```

Comment:
My view is that this information is really internal information about the `AgentSet` and shouldn't be visible outside of the package. I don't want extensions to know about it because I think some extension authors would use it instead of `count` (for instance). While `count` always returns an accurate result, the result returned by this can be wildly inaccurate (and I consider the amount of inaccuracy likely to change in the future). 

If extension authors want access, they can use reflection to get access to this method. While I would discourage them from doing so, it puts a big warning label on top of this method saying "this is subject to change or removal" and forces them to account for that in their code.